misc/scripts/download-local.sh

```diff
@@ -0,0 +1,543 @@
+    if [[ $entry != *::* && $entry == *#*=* ]]; then
+        dest="${url%%#*}"
+        dest="${dest##*/}"
+    fi
+    case $url in
+        *#branch=*)
+            git_branch="${url##*#branch=}"
+            url="${url%%#branch=*}"
+            ;;
+        *#tag=*)
+            git_tag="${url##*#tag=}"
+            url="${url%%#tag=*}"
+            ;;
+        *#commit=*)
+            git_commit="${url##*#commit=}"
+            url="${url%%#commit=*}"
+            ;;
+    esac
+    url="${url%%#*}"
+    if [[ $entry == *::* ]]; then
+        dest="${entry%%::*}"
+    elif [[ $entry != *#*=* ]]; then
+        url="$entry"
+        dest="${url##*/}"
+    fi
+    if [[ ${dest} == *"?"* ]]; then
+        dest="${dest%%\?*}"
+    fi
+}
+
+function calc_git_pkgver() {
+    unset comp_git_pkgver
+    local calc_commit
+    if [[ $url == git+* ]]; then
+        url="${url#git+}"
+    fi
+    if [[ -n ${git_branch} ]]; then
+        calc_commit="$(git ls-remote "${url}" "${git_branch}")"
+    elif [[ -n ${git_tag} ]]; then
+        calc_commit="$(git ls-remote "${url}" "${git_tag}")"
+    elif [[ -n ${git_commit} ]]; then
+        calc_commit="${git_commit}"
+    else
+        calc_commit="$(git ls-remote "${url}" HEAD)"
+    fi
+    comp_git_pkgver="${calc_commit:0:8}"
+}
+
+function genextr_declare() {
+    unset ext_method ext_deps
+    # shellcheck disable=SC2031,SC2034
+    case "${url,,}" in
+        *.zip)
+            ext_method="unzip -qo"
+            ext_deps=("unzip")
+            ;;
+        *.tar.gz | *.tgz)
+            ext_method="tar -xzf"
+            ext_deps=("tar" "gzip")
+            ;;
+        *.tar.bz2 | *.tbz2)
+            ext_method="tar -xjf"
+            ext_deps=("tar" "bzip2")
+            ;;
+        *.tar.xz | *.txz)
+            ext_method="tar -xJf"
+            ext_deps=("tar" "xz-utils")
+            ;;
+        *.tar.zst | *.tzst)
+            ext_method="tar -xf"
+            ext_deps=("tar" "zstd")
+            ;;
+        *.gz)
+            ext_method="gunzip"
+            ext_deps=("gzip")
+            ;;
+        *.bz2)
+            ext_method="bunzip2"
+            ext_deps=("bzip2")
+            ;;
+        *.xz)
+            ext_method="unxz"
+            ext_deps=("xz-utils")
+            ;;
+        *.lz)
+            ext_method="lzip -d"
+            ext_deps=("lzip")
+            ;;
+        *.lzma)
+            ext_method="unlzma"
+            ext_deps=("xz-utils")
+            ;;
+        *.zst)
+            ext_method="unzstd -q"
+            ext_deps=("zstd")
+            ;;
+        *.7z)
+            ext_method="7za x"
+            ext_deps=("p7zip-full")
+            ;;
+        *.rar)
+            ext_method="unrar x -inul"
+            ext_deps=("unrar")
+            ;;
+        *.lz4)
+            ext_method="lz4 -d"
+            ext_deps=("liblz4-tool")
+            ;;
+        *.tar)
+            ext_method="tar -xf"
+            ext_deps=("tar")
+            ;;
+    esac
+}
```

Comment:
Ok so I understand why you're doing this but I don't like conditionally adding dependencies to a deb based on if someone doesn't have that compression tool installed. I know that arch uses `bsdtar` for all their archiving purposes, so we could use that by just adding `libarchive-tools` to the install process to install it.

Reply:
> but I don't like conditionally adding dependencies to a deb based on if someone doesn't have that compression tool installed

The dependencies are added in any case if the archive format requires them. Doesn't matter if they're already installed or not

Reply:
Correct, what vig said